### PR TITLE
satellite/audit: do not contain nodes for unknown errors

### PR DESCRIPTION
**Pull request by mobyvb**
What: 
In some of the cases during audit (Verify + Reverify), skip nodes instead of containing them. We only want to contain nodes when the connection closes or times out during the request. In other error cases we just want to skip them.

Also adds verify/reverify related monkit stats to the lockfile.

Why:
We do not want to risk disqualifying nodes for bugs that are not their fault. Until we are confident that there are no bugs in our code, we should not penalize nodes for unknown errors. This PR will allow us to reenable disqualification and feel confident that any disqualified nodes deserve to be disqualified.
https://storjlabs.atlassian.net/browse/V3-3024

Please describe the tests:
All four tests are nearly identical. So even though a lot of lines are added, it should be easy to review all the tests after seeing one. A "bad blobs store" has been added to facilitate in testing the unknown error case. It behaves exactly like a normal blob store, except when an error is set, in which case it returns that error for any blob store function call.
 - Test 1: `TestVerifierSlowDownload` uploads a file to four storagenodes. One node's database is slowed down beyond what the timeout set on the satellite will allow. We expect after calling `Verify` that the slow node is in the `Contained` list while the remaining three nodes are in the `Successful` list.
 - Test 2: `TestVerifierUnknownError` uploads a file to four storagenodes. One node's database is modified to return an error when the satellite requests data. We expect after calling `Verify` that the bad node is in the `Unknown` list while the remaining three nodes are in the `Successful` list.
 - Test 3: `TestReverifySlowDownload` uploads a file to four storagenodes. One node's database is slowed down beyond what the timeout set on the satellite will allow. The slow node is added to containment mode. We expect after calling `Reverify` that the slow node is in the `Contained` list.
 - Test 4: `TestVerifierUnknownError` uploads a file to four storagenodes. One node's database is modified to return an error when the satellite requests data. The bad node is added to containment mode. We expect after calling `Reverify` that the bad node is in the `Unknown` list.
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: private/testblobs/bad.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+// Delete deletes the blob with the namespace and key.
+func (bad *BadBlobs) Delete(ctx context.Context, ref storage.BlobRef) error {
+	if bad.err != nil {
+		return bad.err
+	}
+	return bad.blobs.Delete(ctx, ref)
+}
+
+// DeleteWithStorageFormat deletes the blob with the namespace, key, and format version
```
Comment (crawter) [inline private/testblobs/bad.go]: dot is missed in the end of sentence

File: private/testblobs/bad.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+func (bad *BadBlobs) DeleteWithStorageFormat(ctx context.Context, ref storage.BlobRef, formatVer storage.FormatVersion) error {
+	if bad.err != nil {
+		return bad.err
+	}
+	return bad.blobs.DeleteWithStorageFormat(ctx, ref, formatVer)
+}
+
+// Stat looks up disk metadata on the blob file
```
Comment (crawter) [inline private/testblobs/bad.go]: dot is missed in the end of sentence

File: satellite/audit/reverify_test.go [hunk 3]
```diff
@@ -955,3 +960,192 @@ func TestReverifyExpired2(t *testing.T) {
...
+		path, err := queue.Next()
+		require.NoError(t, err)
+
+		pointer, err := satellite.Metainfo.Service.Get(ctx, path)
+		require.NoError(t, err)
+
+		slowPiece := pointer.Remote.RemotePieces[0]
+		slowNode := slowPiece.NodeId
+
+		randomIndex, err := audit.GetRandomStripe(ctx, pointer)
```
Comment (navillasa) [inline satellite/audit/reverify_test.go]: nit/not unique to this test - this line could maybe be moved closer to line 1024 where `randomIndex` is used

File: satellite/audit/reverify_test.go [hunk 4]
```diff
@@ -955,3 +960,192 @@ func TestReverifyExpired2(t *testing.T) {
...
+
+		audits.Chore.Loop.TriggerWait()
+		path, err := queue.Next()
+		require.NoError(t, err)
+
+		pointer, err := satellite.Metainfo.Service.Get(ctx, path)
+		require.NoError(t, err)
+
+		slowPiece := pointer.Remote.RemotePieces[0]
+		slowNode := slowPiece.NodeId
```
Comment (navillasa) [inline satellite/audit/reverify_test.go]: nit/personal preference - could be cool to have a comment here saying "select one node to be slow" just to distinguish it from all the other setup happening.

**Pull request merged**